src/main/resources/assets/admin/common/js/form/set/itemset/FormItemSetOccurrenceView.ts

```diff
@@ -62,25 +63,34 @@ export class FormItemSetOccurrenceView
         return this.getFormSet().getFormItems();
     }
 
-    private setLabel() {
-        const firstNonEmptyInput = $(this.formSetOccurrencesContainer.getHTMLElement())
-            .find('.input-wrapper input, .input-wrapper textarea').toArray()
-            .find(input => {
-                const isInput = input.nodeName === 'INPUT';
-                const value = isInput ? (<HTMLInputElement>input).value : StringHelper.htmlToString(input['value']);
-                return value.trim().length > 0;
+    protected getLabelText(): string {
+        const propArrays = this.propertySet.getPropertyArrays();
+        const selectedValues = [];
+
+        if (propArrays && propArrays.length > 0) {
+            propArrays.some((propArray: PropertyArray) => {
+                this.recursiveFetchLabels(propArray, selectedValues, true);
+                return selectedValues.length > 0;
             });
+        }
 
-        if (firstNonEmptyInput) {
-            const isInput = firstNonEmptyInput.nodeName === 'INPUT';
-            if (isInput) {
-                this.label.setText((<HTMLInputElement>firstNonEmptyInput).value);
-            } else {
-                this.label.setText(StringHelper.htmlToString(firstNonEmptyInput['value']));
-            }
-            this.formSetOccurrencesContainer.unDescendantAdded();
+        if (selectedValues.length === 0) {
+            return this.getFormSet().getLabel();
         } else {
-            this.label.setText(this.getFormSet().getLabel());
+            return selectedValues[0];
         }
     }
+
+    private recursiveFetchLabels(propArray: PropertyArray, labels: string[], firstOnly?: boolean): void {
+        propArray.forEach((prop: Property) => {
+            if (ValueTypes.STRING.equals(prop.getType()) && prop.getValue().isNotNull()) {
+                labels.push(prop.getString());
+                if (firstOnly) {
+                    return;
+                }
```

Comment:
`return;`

Reply:
yes, because the method returns void and we want to quit.